src/main/java/com/commercetools/sync/types/utils/FieldDefinitionUpdateActionUtils.java

```diff
@@ -0,0 +1,110 @@
+package com.commercetools.sync.types.utils;
+
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.models.LocalizedString;
+import io.sphere.sdk.types.EnumFieldType;
+import io.sphere.sdk.types.FieldDefinition;
+import io.sphere.sdk.types.LocalizedEnumFieldType;
+import io.sphere.sdk.types.Type;
+import io.sphere.sdk.types.commands.updateactions.ChangeFieldDefinitionLabel;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static com.commercetools.sync.commons.utils.CommonTypeUpdateActionUtils.buildUpdateAction;
+import static com.commercetools.sync.types.utils.TypeUpdateLocalizedEnumActionUtils.buildLocalizedEnumValuesUpdateActions;
+import static com.commercetools.sync.types.utils.TypeUpdatePlainEnumActionUtils.buildEnumValuesUpdateActions;
+import static java.util.stream.Collectors.toList;
+
+
+public final class FieldDefinitionUpdateActionUtils {
+
+    /**
+     * Compares all the fields of an {@link FieldDefinition} and an {@link FieldDefinition} and returns
+     * a list of {@link UpdateAction}&lt;{@link Type}&gt; as a result. If both the {@link FieldDefinition}
+     * and the {@link FieldDefinition} have identical fields, then no update action is needed and hence an
+     * empty {@link List} is returned.
+     *
+     * @param oldFieldDefinition  the field definition which should be updated.
+     * @param newFieldDefinition the field definition draft where we get the new fields.
+     * @return A list with the update actions or an empty list if the field definition fields are identical.
+     */
+    @Nonnull
+    public static List<UpdateAction<Type>> buildActions(
```

Comment:
There is no update actions for type and reference changes in the API.  What should I do?

Reply:
@lojzatran can you please elaborate more on what you mean exactly by those changes with maybe an example?

Reply:
@ahmetoz you are handling type changes [here](https://github.com/commercetools/commercetools-sync-java/pull/304/files#diff-2a49b3ef86c96b2067d1214d267a3237R166) by the work around of (remove-then-add)

Reply:
If you have a Set of enums and you want to add new enum to this Set, will it work? Set of enums like this:
```
{
  "name": "setOfEnums",
  "label": {
    "en": "setOfEnums"
  },
  "isRequired": false,
  "type": {
    "name": "set",
    "elementType": {
      "name": "enum",
      "values": [
        {
          "key": "key1",
          "label": "label1"
        },
        {
          "key": "key2",
          "label": "label2"
        }
      ]
    }
  },
  "attributeConstraint": "None",
  "isSearchable": true,
  "inputHint": "SingleLine",
  "displayGroup": "Other"
}
```

Reply:
You mean a new enum value? You are right! then with this implementation it won't be able to apply the changes (add/remove/change order of enum values) on either a `Set` of `EnumType` values or a `Set` of `LocalizedEnumType` field Definition. 

@ahmetoz in the documentation (https://docs.commercetools.com/http-api-projects-types.html#add-localizedenumvalue-to-fielddefinition) it actually mentions explicitly that the update actions work also on a `Set` of either:

> It can update a `LocalizedEnumType` field definition or a `Set` of `LocalizedEnumType` field definition.


So I guess it should be added to the implementation. It is also missing from the implementation of the product type sync! I will create an issue for it. 

@ahmetoz I would recommend you to address it in a separate PR based off of this one, so that the review is easier. 

@lojzatran Can you also give an example by what you meant with `ReferenceType`?

Reply:
I created an issue here: https://github.com/commercetools/commercetools-sync-java/issues/313

Reply:
`ReferenceType` I'm not sure, it was just a question 😄 I've never worked with `ReferenceType` before, so I cannot say much how would it work. But I saw it was mentioned in the docs, so I asked a question here. Someone has to investigate more deeply in this topic.

Reply:
@ahmetoz can you please check that? :) Especially in the case of `NestedTypes`